### DISCUSSION TITLE
I'm facing some lag while typing the callsign

### DISCUSSION THREAD

**Discussion by KS5Y** [mbridak/not1mm#87] [General]
Hi folks,
Thanks for not1mm, I really appreciate the effort on this.

I noticed some lag while typing the callsign, it seems like it's trying to find the country and heading while I'm typing.

To enter a callsign it takes about 5 seconds, I don't think this is normal.
is there a way to help with this in my workstation? maybe remove the callsign lookup?

Thanks

**Comment by KS5Y** [comment]
Sorry, forgot to add, I'm using version 24.3.16 on FreeBSD 13.1 KDE5

Thanks

**Comment by mbridak** [comment]
Hmm. I'm not able to replicate any noticeable lag. I've turned on everything I could, QRZ lookups all the other windows etc. My laptop is not especially good either. It's a 12yr old I5. I'm not sure what's causing your issue.
 
[Screencast from 2024-03-19 12-46-49.webm](https://github.com/mbridak/not1mm/assets/49108421/a53ea59d-3263-4b3d-a1ca-d6b16c5e4cdc)

**Comment by mbridak** [comment]
For giggles I'm currently trying to install FreeBSD into a VM to see if I can gleen some nuggets of knowledge.
The installer is giving me vibes of 1998 so far.
![Screenshot from 2024-03-19 14-57-17](https://github.com/mbridak/not1mm/assets/49108421/8d876538-59fc-4391-929a-723d60c0befe)

**Reply by mbridak** [reply]
Base layer installed.
added myself to the video group.
managed to install Xorg...
startx command works.
Wow! This reminds me of when I used to use CDE.
Currently trying to install KDE Plasma5.
Crossing fingers and toes.

**Reply by mbridak** [reply]
![Screenshot from 2024-03-19 15-41-12](https://github.com/mbridak/not1mm/assets/49108421/5338bf20-58bf-4a7c-8458-c006c77a461a)

Okay... Base install comes with Python 3.9... Eww, but okay... Gotta get pip installed.

**Reply by mbridak** [reply]
Okay making progress.
Figured out I didn't add myself to wheel group.
Couldn't login as root to kde, couldn't kill xorg cause I wasn't root, couldn't disable sddm same.
booted into single user mode. Scratched head till I figured out how to mount the drive RW.
Added myself to wheel.
Got pip installed. Couldn't pip install PyQt5 cause no one can.
 Found pkg for it.  
ran out of time gotta pick up son.

**Reply by mbridak** [reply]
![Screenshot from 2024-03-19 17-16-12](https://github.com/mbridak/not1mm/assets/49108421/a9702e9b-b0ca-4d88-b04f-be9d55d3197b)

Okay it's alive. And YES! Yes it does suck on FreeBSD. I wonder why... I'll have to see if I can poke around. I'll tell you if I find anything.

**Reply by mbridak** [reply]
Okay, it's the super check partial routine.
There's good news and bad news.

The good news is I can completely cut that routine from the loggers main window since it has no actual use for it, since that has been moved to the check window. And the check window is running the same call a second time. this will speed up your main window. 

The bad news is the check window will still lag. I don't know if this is because the python version is so old or not.

Either way I'll rip the useless bloat out and post a new version for you.

**Reply by KS5Y** [reply]
@mbridak WOW! you did all that in few time, I have been testing not1mm for some time now. 

what version of python are you running in Linux?

I can try some updates.

Thank you so much

**Reply by mbridak** [reply]
I'm running Python 3.12 They've sped it up quite a bit since 3.9

I gutted the useless code and pushed a new version to github and pypi.
You should be able to update and notice a pretty good speed bump in the interface.
Hope that helps! It's good you mentioned it. I would not have ever caught that otherwise.

**Reply by KS5Y** [reply]
@mbridak Thank you so much that did the trick!!

**Reply by KS5Y** [reply]
Also, I have issues with "Bandmap" do you want me to create a new thread?

Thanks

**Reply by mbridak** [reply]
Excellent! Have a good one.

Mike K6GTE 73!
